jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1779,6 +1779,9 @@ public static URI correctURI(URI uri)
         if (colon < 0)
             return uri; // path portion not found
 
+        if ((colon + 2) == rawURI.length())
```

Comment:
Maybe >=? I believe this method shouldn't throw, even on 'file:' or something malformed like this. Or if it throws, it should throw `MalformedURLException`. See `ServletContextHandler#getResource` contract. And an implementation is calling this method.

Reply:
this is just testing for the exact input of `file:/`